### ISSUE TITLE
Multi control plane node creation failing with podman

### ISSUE THREAD

**Issue by aadi58002** [kubernetes-sigs/kind#2996]
**What happened**:
The cluster creation fails when trying to create multi control plane node cluster. ( I am using the default example config on the site [here](https://kind.sigs.k8s.io/docs/user/quick-start/#controlplane-ha))
**What you expected to happen**:
Create six nodes with three being the control plane and 3 being worker nodes
**How to reproduce it (as minimally and precisely as possible)**:
I am using podman to run containers.

**Anything else we need to know?**:
Error states that the it is related to kubeadm failing.
link to the error : https://pastebin.com/QYiqvdxj

It seems to be related to a previous issue as well #2880 
**Environment:**

- kind version: (use `kind version`): kind v0.16.0 go1.19.1 linux/amd64
- Kubernetes version: (use `kubectl version`): Client Version: v1.25.3, Kustomize Version: v4.5.7
- Docker version: (use `docker info`): podman version 4.3.0
- OS (e.g. from `/etc/os-release`): Arch Linux

**Comment by BenTheElder** [comment]
> Create six nodes with three being the control plane and 3 being worker nodes

There are a bunch of dimensions on which you can hit artificial limits constraining your ability to run multi node clusters, for example: https://kind.sigs.k8s.io/docs/user/known-issues/#pod-errors-due-to-too-many-open-files

Can you share `kind create cluster --retain --config=your-config && kind export logs && kind delete cluster` / upload those logs, and can you check what the smallest failure case is?

**Comment by BenTheElder** [comment]
What we can tell from the existing logs is the api server didn't come up healthy, but to know why we need the rest of the logs from `kind export logs` (with `--retain` to prevent cleanup).

Does a single node cluster work?

Also, while that config illustrates what is possible, it is rare that making a cluster of this shape makes sense. These nodes share a host kernel and CPU / Memory and few applications have a valid use case for testing on more than one node.

**Comment by aadi58002** [comment]
Thanks for the reply

> kind create cluster --retain --config=your-config && kind export logs && kind delete cluster

The failure starts occuring when i add a second control plane node (i tested with 9 worker nodes and 1 control plane and it works fine then.
But even when working with 2 control plane node it fails.
![Screenshot_20221109_100344](https://user-images.githubusercontent.com/96418027/200739430-747e3dab-d435-4c1d-b2dd-afa0ef239298.png)

 
> What we can tell from the existing logs is the api server didn't come up healthy, but to know why we need the rest of the logs from `kind export logs` (with `--retain` to prevent cleanup).
[logs.zip](https://github.com/kubernetes-sigs/kind/files/9967475/logs.zip)
> 
> Does a single node cluster work?
>> Yes , i didn't ever have a problem in case of single control plane node clusters.
> 
> Also, while that config illustrates what is possible, it is rare that making a cluster of this shape makes sense. These nodes share a host kernel and CPU / Memory and few applications have a valid use case for testing on more than one node.
>> I am not using it to simulate production environment. I am currently learning the devops field and i heard it said      >> many times that it is good to have redundancy with the control plane as if one would fail others can take up control >> and prevent malfunctions.

**Comment by BenTheElder** [comment]
For local development purposes with kind multiple control plane nodes doesn't improve redundancy in any meaningful way, it's useful if you're testing kubernetes' control plane internals though.

Sounds like it may be an issue with the control plane loadbalancer. Will inspect the logs later when I get some more time.

**Comment by aojea** [comment]
> Sounds like it may be an issue with the control plane loadbalancer. Will inspect the logs later when I get some more time.

it is 

> Nov 09 04:53:46 kind-control-plane kubelet[260]: E1109 04:53:46.201710     260 reflector.go:140] vendor/k8s.io/client-go/informers/factory.go:134: Failed to watch *v1.CSIDriver: failed to list *v1.CSIDriver: Get "https://kind-external-load-balancer:6443/apis/storage.k8s.io/v1/csidrivers?limit=500&resourceVersion=0": EOF

**Comment by BenTheElder** [comment]
Hmm the logs zip doesn't even have anything from the loadbalancer container ...

**Comment by aadi58002** [comment]
> Hmm the logs zip doesn't even have anything from the loadbalancer container ...

- kind export logs
Exported a folder to a dir in /tmp/(something) I just zipped it and send it here I will look into how to get the load balancer loga

**Comment by BenTheElder** [comment]
Yeah, that suggests for some reason the container doesn't exist at all?

**Comment by BenTheElder** [comment]
To be quite honest I would recommend using docker. The podman backend is still experimental due to ongoing compatibility issues, even podman vs previous versions of podman is still regularly breaking for us.

**Comment by BenTheElder** [comment]
It's not clear why there are no logs for the loadbalancer node, even if it failed to run the container should have been around and had container level logs. Is it possible something else deleted it?

We don't have enough info to debug what is wrong with that container currently, but also this container is very simple and there's not a good reason it shouldn't work under podman.

Multi-control-plane mode is not really prioritized at the moment, Kubernetes is not testing it much and there are many other things ... Podman is varyingly incompatible and also on the backburner for kind, unfortunately I won't have time to attempt to reproduce this anytime soon.

If you can find a minimum reproducer and identify if this container even runs and what the logs are, we can look into it further. 

> It seems to be related to a previous issue as well https://github.com/kubernetes-sigs/kind/issues/2880

This is just another issue with the same symptom, that kubeadm fails. Which can happen for many reasons.
The previous issue is not related, they're creating a single node.
